networkx/algorithms/approximation/traveling_salesman.py

```diff
@@ -34,6 +34,10 @@
 """
 import math
 import networkx as nx
+import numpy as np
+import scipy.optimize as sp
+
+from networkx import NetworkXNoCycle
```

Comment:
In order to make it possible for users without numpy to use most of NetworkX, we put the `import numpy` line inside the function that uses numpy.

For the scipy.optimize import, I think this should be `as optimize` instead of `as sp`.  The name `sp` is usually reserved for scipy itself.  You could use `opt` instead of `optimize` or similar. But don't use `sp`.

And instead of importing `NetworkXNoCycle` you should probably just use `nx.NetworkXNoCycle`.